File: fingertip/src/main/java/com/praszapps/fingertip/presenter/FingertipActivityPresenter.java
package com.praszapps.fingertip.presenter;

import com.praszapps.fingertip.MVP.FingertipMVPContract;
import com.praszapps.fingertip.model.repository.FingertipRepository;

import javax.inject.Inject;
import javax.inject.Named;

public class FingertipActivityPresenter implements FingertipMVPContract.Presenter {

    @Inject
    @Named("activityView")
    public FingertipMVPContract.View view;

    @Inject
    FingertipRepository repository;

    @Override
    public void authenticateViaFingerprint() {

    }

    @Override
    public void stopFingerprintTracking() {

    }
}

File: fingertip/src/main/java/com/praszapps/fingertip/presenter/FingertipActivityPresenter.kt
package com.praszapps.fingertip.presenter

import com.praszapps.fingertip.MVP.FingertipMVPContract
import com.praszapps.fingertip.model.repository.FingertipRepository

import javax.inject.Inject
import javax.inject.Named

class FingertipActivityPresenter : FingertipMVPContract.Presenter {

    @Inject
    @Named("activityView")
    lateinit var view: FingertipMVPContract.View

    @Inject
    lateinit var repository: FingertipRepository

    override fun authenticateViaFingerprint() {

    }

    override fun stopFingerprintTracking() {

    }
}
